tealiumlibrary/src/main/java/com/tealium/core/Tealium.kt

```diff
@@ -316,6 +317,57 @@ class Tealium @JvmOverloads constructor(val key: String, val config: TealiumConf
      */
     @Suppress("unused")
     fun killTraceVisitorSession() {
-        activityObserverListener.killTraceVisitorSession()
+        deepLinkHandler.killTraceVisitorSession()
+    }
+
+    /**
+     * Migrates persistent data from the Tealium Android (Java) library if present
+     * */
+    private fun migratePersistentData() {
+        val hashCode = (config.accountName + '.' +
+                    config.profileName + '.' +
+                    config.environment.environment).hashCode()
+        val legacySharedPreferences = config.application.getSharedPreferences("tealium.datasources.${Integer.toHexString(hashCode)}", 0)
+        if (legacySharedPreferences.all.isEmpty()) {
+            return
+        }
+        legacySharedPreferences.all.forEach { item ->
+            val key = item.key
+            val value = item.value
+
+            (value as? String)?.let {
+                dataLayer.putString(key, it, Expiry.FOREVER)
+            }
+            (value as? Boolean)?.let {
+                dataLayer.putBoolean(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Double)?.let {
+                dataLayer.putDouble(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Int)?.let {
+                dataLayer.putInt(key, it, Expiry.FOREVER)
+            }?:
+            (value as? Long)?.let {
+                dataLayer.putLong(key, it, Expiry.FOREVER)
+            }?:
+            if (value is Set<*>) {
+                value.firstOrNull()?.let {
```

Comment:
not sure of the need for `firstOrNull` as would that not get (and only get) the first entry of the set